sqlglot/optimizer/qualify_columns.py

```diff
@@ -352,6 +375,22 @@ def _add_replace_columns(expression, tables, replace_columns):
         replace_columns[id(table)] = columns
 
 
+def _add_pivot_columns(pivot, source_columns, columns):
+    pivot_output_columns = [col.output_name for col in pivot.args.get("columns", [])]
+    if not pivot_output_columns:
+        pivot_output_columns = [col.alias_or_name for col in pivot.expressions]
```

Comment:
The parser should always produce this `columns` arg, but this check was added as a sanity check / fallback in case we get an `exp.Pivot` expression that was programmatically produced and doesn't have it set.